apps/webservice/src/app/[workspaceSlug]/(app)/resources/(sidebar)/relationship-rules/components/CreateRelationshipDialog.tsx

```diff
@@ -313,48 +316,68 @@ export const CreateRelationshipDialog: React.FC<
               <FormField
                 control={form.control}
                 name="metadataKeys"
-                render={({ field }) => (
-                  <FormItem>
-                    <FormControl>
-                      <div className="flex flex-wrap gap-2">
-                        {field.value.map((key, index) => (
-                          <div
-                            key={index}
-                            className="flex items-center gap-1 rounded-md bg-secondary px-2 py-1"
-                          >
-                            <span>{key}</span>
-                            <button
-                              type="button"
-                              onClick={() => {
-                                const newKeys = [...field.value];
-                                newKeys.splice(index, 1);
-                                field.onChange(newKeys);
+                render={({ field }) => {
+                  const addKey = () => {
+                    const value = currentMetadataKey.trim();
+                    if (value && !field.value.includes(value)) {
+                      field.onChange([...field.value, value]);
+                      setCurrentMetadataKey("");
+                    }
+                  };
+
+                  return (
+                    <FormItem>
+                      <FormControl>
+                        <div className="flex flex-wrap items-start gap-2">
+                          {field.value.map((key, index) => (
+                            <div
+                              key={index}
+                              className="flex items-center gap-1 rounded-md bg-secondary px-2 py-1"
+                            >
+                              <span>{key}</span>
+                              <button
+                                type="button"
+                                onClick={() => {
+                                  const newKeys = [...field.value];
+                                  newKeys.splice(index, 1);
+                                  field.onChange(newKeys);
+                                }}
+                                className="ml-1 text-muted-foreground hover:text-foreground"
+                              >
+                                ×
+                              </button>
+                            </div>
+                          ))}
+                          <div className="flex min-w-[200px] flex-1 items-center gap-2">
+                            <Input
+                              className="!mt-0 flex-1"
```

Comment:
```suggestion
                              className="flex-1"
```

Seems weird? maybe put padding on the parent div?